hanadb_exporter/main.py

```diff
@@ -105,6 +105,43 @@ def lookup_etc_folder(config_files_path):
     raise ValueError(
         'configuration file does not exist in {}'.format(",".join(config_files_path)))
 
+
+def register_connectors(hosts, current_host, master_host):
+    """
+    True: if the current host is the MASTER host, or if the exporter is running on a standalone host.
+    False: if the current host is a WORKER host.
+    """
+    if current_host == master_host or current_host not in hosts:
+        return True
+    else:
+        return False
+
+
+def start(db_manager, host, port, user, password, userkey, scale_out_mode, multi_tenant, timeout, metrics):
+    db_manager.start(
+        host, port,
+        user=user,
+        password=password,
+        userkey=userkey,
+        multi_tenant=multi_tenant,
+        timeout=timeout)
+
+    connectors = db_manager.get_connectors()
+    collector = prometheus_exporter.SapHanaCollectors(connectors=connectors, metrics_file=metrics)
+    hosts = db_manager.get_database_hosts()
+    LOGGER.info('current_host: ' + utils.get_hostname())
+    if scale_out_mode:
+        LOGGER.info('scale_out_mode mode is enabled')
+        if register_connectors(hosts, current_host=utils.get_hostname(), master_host=host):
```

Comment:
I don't register the connector when the export is running on a worker node, so the call to start again down below will add the connectors in case of failover, so the connectors are added once.